### PR TITLE
SQL: (0.4) TRX_ID support in SELECT

### PR DESCRIPTION
**Pull request by midenok**
Please, review.

This should be merged after 0.3 is finished.

### CODE REVIEW

File: sql/sql_select.cc [hunk 1]
```diff
@@ -673,7 +673,7 @@ setup_without_group(THD *thd, Ref_ptr_array ref_pointer_array,
 }
 
 static int
-setup_for_system_time(THD *thd, TABLE_LIST *tables, COND **where_expr, SELECT_LEX *select_lex)
+setup_for_system_time(THD *thd, TABLE_LIST *tables, COND **where_expr, SELECT_LEX *slex)
```
Comment (kevgs) [inline sql/sql_select.cc:676]: `select_lex` is a better name, IMO. There is no well-known `slex`. By well-known I mean `i`, `j`, `k`, `buf`, etc.

File: sql/sql_yacc.yy [hunk 2]
```diff
@@ -8870,23 +8871,45 @@ opt_for_system_time_clause:
...
           TIMESTAMP simple_expr
           AND_SYM
           TIMESTAMP simple_expr
           {
-            Lex->current_select->vers_conditions.init(FOR_SYSTEM_TIME_BETWEEN, $4, $7);
+            Lex->current_select->vers_conditions.init(FOR_SYSTEM_TIME_BETWEEN, UNIT_TIMESTAMP, $4, $7);
+          }
+        | FOR_SYSTEM_TIME_SYM
+          AS OF_SYM
+          TRX_ID_SYM simple_expr
```
Comment (kevgs) [inline sql/sql_yacc.yy:8894]: Why do we need it, for trx_id arithmetic, or something like `CONVERT_DATE_TO_TRX_ID()`? 
And if we need it here what I'll get if I put here an `"arbitrary string"`, a bogus result or warning/error?

**Review by kevgs — COMMENTED** [review]
No tests. I think it's always better to add passing tests along with the feature.

File: sql/sql_select.cc (hunk 1 above)
Comment (midenok) [reply sql/sql_select.cc:676]: It's func arg, so it doesn't matter. I shortened it because of too long expressions.

File: sql/sql_yacc.yy (hunk 2 above)
Comment (midenok) [reply sql/sql_yacc.yy:8894]: Let me check. It just returned empty result set with 2 warnings:

```
+---------+------+-----------------------------------------+
| Level   | Code | Message                                 |
+---------+------+-----------------------------------------+
| Warning | 1292 | Truncated incorrect DOUBLE value: 'xyz' |
| Warning | 1292 | Truncated incorrect DOUBLE value: 'xyz' |
+---------+------+-----------------------------------------+
```

**Comment by midenok** [comment]
> No tests. I think it's always better to add passing tests along with the feature.

Agree. Will make tests in different commit and update. I would ask you to push tests in separate commit for convenience as well.

**Comment by kevgs** [comment]
Fine. For a I put a note about separating feature code and tests. https://github.com/natsys/mariadb_10.2/wiki

Patch looks good to me.

**Pull request closed without merging**